src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Undoes the previous add, delete or edit command
+ */
+public class UndoCommand extends Command {
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_SUCCESS = "Undo %s command";
+
+    public static final String MESSAGE_NO_COMMAND_TO_UNDO = "No add, delete or edit commands in the"
+            + " command history to undo";
+
+    public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command in command history: %s";
+
+    private static Stack<String> commandHistory = new Stack<>();
+
+    private static final Pattern BASIC_COMMAND_FORMAT =
+            Pattern.compile("(?<commandWord>\\S+(\\s(appointment|property|all))?)(?<arguments>.*)");
+
+    /**
+     * Logs the command string {@code commandString} to the command history stack.
+     */
+    public static void logCommand(String commandString) {
```

Comment:
Can this method be shifted to `PocketEstateParser` instead? So that the regex and parsing are not duplicated. 
Let me know if it is not possible because it will affect the implementation or if there are any other reasons

Reply:
I have added `getCommandWord(String)` in `PocketEstateParser`